File: api_bus/core/models/horario.py
from django.db import models
from .ruta import Ruta

class Horario(models.Model):
    DIAS_CHOICES = [
        ('Lunes', 'Lunes'),
        ('Martes', 'Martes'),
        ('Miércoles', 'Miércoles'),
        ('Jueves', 'Jueves'),
        ('Viernes', 'Viernes'),
        ('Sábado', 'Sábado'),
        ('Domingo', 'Domingo'),
    ]
    
    ruta = models.ForeignKey(Ruta, on_delete=models.CASCADE, related_name='horarios')
    hora_salida = models.TimeField()
    hora_llegada = models.TimeField()
    dias_disponibles = models.JSONField()  # Almacena array de días
    capacidad_disponible = models.PositiveIntegerField()
    
    def __str__(self):
        return f"{self.ruta.nombre} - {self.hora_salida}"

File: api-v2/models/horario.js
// models/horario.js (ESM)
import { DataTypes } from 'sequelize';
import sequelize from '../config/db.js';

const Horario = sequelize.define(
  'Horario',
  {
    // en cada modelo:
id: {
  type: DataTypes.BIGINT.UNSIGNED,
  autoIncrement: true,
  primaryKey: true,
},

    rutaId: {
      type: DataTypes.BIGINT.UNSIGNED,
      allowNull: true,
      references: { model: 'Rutas', key: 'id' },
      onUpdate: 'CASCADE',
      onDelete: 'SET NULL',
    },
    hora_salida: {
      type: DataTypes.TIME,
      allowNull: true,
    },
    hora_llegada: {
      type: DataTypes.TIME,
      allowNull: true,
    },
    dias_disponibles: {
      type: DataTypes.JSON, // array de días o null
      allowNull: true,
    },
    capacidad_disponible: {
      type: DataTypes.INTEGER.UNSIGNED,
      allowNull: true,
    },
  },
  {
    tableName: 'Horarios',
    timestamps: true,
    underscored: true,
  }
);

Horario.associate = (models) => {
  Horario.belongsTo(models.Ruta, { foreignKey: 'rutaId', as: 'ruta' });
};

export default Horario;
